session.go

```diff
@@ -240,7 +241,7 @@ func (s *Session) recv() {
 	var err error
 	for {
 		err = s.recvMsg()
-		if errors.Is(err, io.EOF) || errors.Is(err, io.ErrUnexpectedEOF) {
+		if isProbableEOF(err) {
```

Comment:
I have two thoughts here.

## All _transport_ errors are unrecoverable
Split message receiving and msg parsing into two seperate methdods `recvMsg` and `parseMsg`   If we have any error from the underlying transport that we consider it a failure and close.  In the future we could perhaps had some sort of special marker to the transport package to indicate a a temporary/retryable error but I can't think of a case where that would be beneficial.
   
   ParseMsg is really where we want to retry (I think).  If we don't know what a message is we can wait for the next one.  Although one could argue that we could bail there as well.   I need to read the RFC a bit again and see if it provides guidance. 


## All _network_ errors are unrecoverable

A different way to handle this would just to match on anything that satisfies the [`net.Error`](https://pkg.go.dev/net#Error) interface and break out of the for loop in that case.  Although I am leaning more to the option above as it will make sure that custom transports or transport specific errors don't have to follow specific error handling to be dealt with appropriately.  Those types of systems can become very fragile over time (case in point the issue you are having).

Reply:
Let me know if this makes sense.  I can help discuss these a bit as well if you want to.  I am just happy to have more people using the library and finding issues!